src/cargo/util/auth/mod.rs

```diff
@@ -284,10 +197,8 @@ fn registry_credential_config_inner(
         ))
     };
     Ok(
-        match (token, credential_process, secret_key, secret_key_subject) {
-            (Some(_), Some(_), _, _) => return err_both("token", "credential-process"),
+        match (token, credential_provider, secret_key, secret_key_subject) {
             (Some(_), _, Some(_), _) => return err_both("token", "secret-key"),
-            (_, Some(_), Some(_), _) => return err_both("credential-process", "secret-key"),
```

Comment:
Just checking, why were these removed?  IIUC, if you have `credential-providers` and a `token` or `secret-key` that it no longer generates an error, and ignores credential-providers? I'm concerned that could be confusing, or make it difficult to start using credential-providers. Would it be possible to retain the previous behavior?

Reply:
I intended it to prefer `credential-provider` if both were set. I'll fix that. More tests are needed.

The intent of the change here was to:
* Allow a `credential-provider` such as `cargo:token`, which would then read corresponding `token` field
* Use the `credential-provider` (for when running with nightly), but then still have the `token` fallback so stable continues to work.